scripts/qa_crowbarsetup.sh

```diff
@@ -4418,6 +4418,14 @@ function oncontroller_testpostupgrade
         scp cirros@$fip:/var/log/ping_outside.out ping_outside.$fip.out
         max=$(sed -n 's/^.* not available for: //p' ping_outside.$fip.out | sort -n | tail -n 1)
         echo "Maximum outage while pinging outside IP from $fip: $max seconds"
+
+        scp cirros@$fip:/mnt/cinder_test.out cinder_test.$fip.out
+        res=$(awk '$1!=p+1{print $1-p}{p=$1}' cinder_test.$fip.out | tail -n +2 | sort | uniq | tail -n 1)
```

Comment:
Let me explain what this line does: the file is filled with a one timestamp per second (in seconds). The `awk` script checks if two consecutive timestamps have a difference of more than one second. If so, calculates the difference and writes that. If there are several gaps in the file, the script will write the difference for each of them.

After the `awk` script, we could consider the output is a list of gaps in seconds. For simplification, the first gap is always `1`, so we skip it with `tail -n +2`.

Then we sort the gaps by decreasing duration (`sort`) eliminating duplicates (`uniq`). We are just going to report the biggest gap, so we get the first in line (`tail -n 1`).

@jsuchome you are right, we might as well remove `uniq`. Will do that.

@Itxaka i hope my explanations covered your question.